hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/spark/sql/hudi/command/index/TestFunctionalIndex.scala

```diff
@@ -557,12 +553,86 @@ class TestFunctionalIndex extends HoodieSparkSqlTestBase {
         checkAnswer(metadataSql)(
           Seq("2020-09-26", "2021-09-26"),
           Seq("2022-09-26", "2022-09-26"),
+          Seq("2022-09-26", "2022-09-26"), // for file in name=a2
           Seq("2024-09-26", "2024-09-26") // for file in name=a3
         )
       }
     }
   }
 
+  // Test functional index using column stats and bloom filters, and then clean older version, and check index is correct.
+  test("Test Functional Index With Cleaning") {
```

Comment:
can we also add a non-partitioned test. 
and a test for MOR table where entire file slice is cleaned up.

Reply:
done.. By `and a test for MOR table where entire file slice is cleaned up`, I am assuming you mean all but the latest file slice in a partition is cleaned up. So, the test is not compacting the table in case of mor so that prev file slice is cleaned up.